src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbBase.java

```diff
@@ -324,12 +330,20 @@ public ListTablesResult listTables(ListTablesRequest listTablesRequest) {
 
     @Override
     public ListTablesResult listTables() {
-        return getAmazonDynamoDb().listTables();
+        if (mtContext.getContextOpt().isEmpty()) {
+            return getAmazonDynamoDb().listTables();
```

Comment:
I think this method would need to be implemented in the subclasses to take prefixes and statically configured tables into account.

Reply:
actually, by filtering with `isMtTable` here, that should be handled for us? That solves both 👍